eng/pipelines/templates/pipeline.yml

```diff
@@ -136,7 +136,7 @@ stages:
       MSBUILDDISABLENODEREUSE: 1
     condition: "and(succeeded(), eq(variables['RunTestsOnLinux'], 'true'))"
     pool:
-      vmImage: ubuntu-latest
+      vmImage: ubuntu-22.04
```

Comment:
The `ububtu-22.04` image is likely to be removed before our release-5.11.x branch goes out of support, so pinning to this image is going to require future maintenance. Obviously we can just deal with it when it breaks, but at that point it becomes a blocker if we have an urgent fix we need to service.

Reply:
That's very fair. This is just kicking the can down the road. I think it makes sense to fix it for now, since we'd rather have 5.11 builds workable than not.

That being said, we do have a tracking issue: https://github.com/NuGet/Client.Engineering/issues/2549, and we could mention it there that our old branches would have the same issue.

On the other hnad, we *may* be able to skip testing build legs that are not shipping in .NET anymore, so we could have solutions for those branches eventually. 
The key part, is should we create an issue? Or just rely on this information here to guide folks in case things do go wrong in the future.